<div align="center"> <b> <span style='color:red'> <h1> IS 362 - 311 SERVICE REQUESTS 2015 </h1> </span> </b> </div>
<div align="center"> <b> <span style='color:green'> <h2> FINAL PROJECT </h2> </span> </b> </div>
<div align="center"> <b> <span style='color:green'> <h4> Name: Rickardo Henry </h4> </span> </b> </div>

For my final project I plan to demonstrate all that I have learned throughout the course. I will be using data from [NYC OpenData](https://opendata.cityofnewyork.us/). Utilizing the available data, I have decided to use [311 Service Requests](https://data.cityofnewyork.us/dataset/311-Service-Requests-From-2015/57g5-etyj) for 2015. I have filtered the data to shorten the amount of wait time for results as the dat is very large. <br /><br /> <ul>With the available data I plan to show:
<li> What was the number #1 service request handled by 311 in 2015? </li>
<li> What season received the most service requests? Name a few of the SR’s </li>
<li> What borough made the most service request in 2015? </li>
<li> What was the total of service requests in 2015? </li>
<li> How many service requests remain open? How many have been closed? </li>
</ul>


I plan to have a pie graph to display my data. I also plan to include some images <i>(if I can figure it out)</i> and some other types of graphs to display more info on service requests for 2015.

<b> We can begin by importing an image and the necessary libraries. </b>

In [5]:
import pandas as pd
import numpy as np
from IPython.display import Image

Image(url='http://www1.nyc.gov/assets/home/images/social/applications/NYC311Logo-fff200-480px-programs.png',
      width=200,
      height=200)

<b> Now we can begin reading the data.</b>

In [6]:
url = 'C:/Users/Rickardo.Henry/Desktop/IS_362_FinalProject/Data%20Files/311_SR_2015_Filtered.csv'
ser_req = pd.read_csv(url)
ser_req.head(10)

Unique Key     Created Date     Closed Date Agency  \
0    31015465   7/6/2015 10:58  7/22/2015 1:07    DCA   
1    30997660   7/3/2015 13:26  7/3/2015 14:08   NYPD   
2    31950223   11/9/2015 3:55  11/9/2015 8:08   NYPD   
3    31000038    7/3/2015 2:18   7/3/2015 7:54   NYPD   
4    30995614    7/4/2015 0:03   7/4/2015 3:33   NYPD   
5    31042454    7/9/2015 0:00  7/20/2015 0:00  DOHMH   
6    31043076   7/9/2015 12:04             NaN    DPR   
7    31037751    7/9/2015 0:00             NaN  DOHMH   
8    31298553  8/12/2015 11:09  8/28/2015 1:06    DCA   
9    31492526   9/9/2015 21:59  9/9/2015 23:17   NYPD   

                               Agency Name                 Complaint Type  \
0           Department of Consumer Affairs             Consumer Complaint   
1          New York City Police Department                        Vending   
2          New York City Police Department               Blocked Driveway   
3          New York City Police Department             Noise - Commercial   
4          New York City Police Department        Noise - Street/Sidewalk   
5  Department of Health and Mental Hygiene                 Standing Water   
6       Department of Parks and Recreation  Root/Sewer/Sidewalk Condition   
7  Department of Health and Mental Hygiene                 Standing Water   
8           Department of Consumer Affairs             Consumer Complaint   
9          New York City Police Department        Noise - Street/Sidewalk   

                    Descriptor                 Location Type  Incident Zip  \
0              Demand for Cash                           NaN       11360.0   
1           In Prohibited Area    Residential Building/House       10019.0   
2                    No Access               Street/Sidewalk       10453.0   
3             Loud Music/Party           Club/Bar/Restaurant       11372.0   
4                 Loud Talking               Street/Sidewalk       11216.0   
5        Other - Explain Below                         Other           NaN   
6  Trees and Sidewalks Program                        Street       10469.0   
7             Puddle in Ground  3+ Family Apartment Building       10016.0   
8      Damaged/Defective Goods                           NaN       11420.0   
9                 Loud Talking               Street/Sidewalk       11238.0   

           Incident Address              City  Status  \
0          27-16 203 STREET           BAYSIDE  Closed   
1    200 CENTRAL PARK SOUTH          NEW YORK  Closed   
2         1993 GRAND AVENUE             BRONX  Closed   
3  84-16 NORTHERN BOULEVARD   JACKSON HEIGHTS  Closed   
4        1057 BERGEN STREET          BROOKLYN  Closed   
5                       NaN               NaN  Closed   
6      3344 PEARSALL AVENUE             BRONX    Open   
7          379 THIRD AVENUE          NEW YORK    Open   
8         127-19 111 AVENUE  SOUTH OZONE PARK  Closed   
9     238 SAINT JAMES PLACE          BROOKLYN  Closed   

                              Resolution Description    Borough  
0  The Department of Consumer Affairs mailed you ...     QUEENS  
1  The Police Department responded to the complai...  MANHATTAN  
2  The Police Department responded to the complai...      BRONX  
3  The Police Department responded to the complai...     QUEENS  
4  The Police Department responded and upon arriv...   BROOKLYN  
5  The Department of Health and Mental Hygiene wi...     QUEENS  
6  The Department of Parks and Recreation usually...      BRONX  
7  The Department of Health and Mental Hygiene wi...  MANHATTAN  
8  The Department of Consumer Affairs mailed you ...     QUEENS  
9  The Police Department responded to the complai...   BROOKLYN

<b> I decided to replace NaN values with '0'.</b> <br /><br /> <i> Only displayed a few rows for easier viewing</i>

In [7]:
ser_req.replace(np.NaN, 0).head(5)

Unique Key    Created Date     Closed Date Agency  \
0    31015465  7/6/2015 10:58  7/22/2015 1:07    DCA   
1    30997660  7/3/2015 13:26  7/3/2015 14:08   NYPD   
2    31950223  11/9/2015 3:55  11/9/2015 8:08   NYPD   
3    31000038   7/3/2015 2:18   7/3/2015 7:54   NYPD   
4    30995614   7/4/2015 0:03   7/4/2015 3:33   NYPD   

                       Agency Name           Complaint Type  \
0   Department of Consumer Affairs       Consumer Complaint   
1  New York City Police Department                  Vending   
2  New York City Police Department         Blocked Driveway   
3  New York City Police Department       Noise - Commercial   
4  New York City Police Department  Noise - Street/Sidewalk   

           Descriptor               Location Type  Incident Zip  \
0     Demand for Cash                           0       11360.0   
1  In Prohibited Area  Residential Building/House       10019.0   
2           No Access             Street/Sidewalk       10453.0   
3    Loud Music/Party         Club/Bar/Restaurant       11372.0   
4        Loud Talking             Street/Sidewalk       11216.0   

           Incident Address             City  Status  \
0          27-16 203 STREET          BAYSIDE  Closed   
1    200 CENTRAL PARK SOUTH         NEW YORK  Closed   
2         1993 GRAND AVENUE            BRONX  Closed   
3  84-16 NORTHERN BOULEVARD  JACKSON HEIGHTS  Closed   
4        1057 BERGEN STREET         BROOKLYN  Closed   

                              Resolution Description    Borough  
0  The Department of Consumer Affairs mailed you ...     QUEENS  
1  The Police Department responded to the complai...  MANHATTAN  
2  The Police Department responded to the complai...      BRONX  
3  The Police Department responded to the complai...     QUEENS  
4  The Police Department responded and upon arriv...   BROOKLYN

<b> I decided to take a look at the data types of each column to understand what types I will be working with and may need to change if necessary. </b>

In [8]:
ser_req.dtypes

Unique Key                  int64
Created Date               object
Closed Date                object
Agency                     object
Agency Name                object
Complaint Type             object
Descriptor                 object
Location Type              object
Incident Zip              float64
Incident Address           object
City                       object
Status                     object
Resolution Description     object
Borough                    object
dtype: object

<b> We now can begin to answer the questions from above. <br /> <br /> 
Q1. What was the #1 service request handled by 311 in 2015? </b>

In [9]:
ser_req['Complaint Type'].value_counts()

Illegal Parking                  29
Blocked Driveway                 24
Noise - Street/Sidewalk          22
Street Condition                 12
Noise - Commercial                9
Consumer Complaint                5
Derelict Vehicle                  4
Overgrown Tree/Branches           4
Damaged Tree                      3
Root/Sewer/Sidewalk Condition     3
Noise - Park                      2
Benefit Card Replacement          2
Standing Water                    2
Homeless Encampment               2
HEAT/HOT WATER                    2
Traffic                           2
Vending                           2
Sidewalk Condition                2
Broken Muni Meter                 2
Noise - Vehicle                   2
DPR Internal                      1
Maintenance or Facility           1
Dirty Conditions                  1
ELECTRIC                          1
Food Establishment                1
FLOORING/STAIRS                   1
Rodent                            1
Dead Tree                   

<b> We can view from the results above that ILLEGAL PARKING complaints was the complaint type with the highest outcome with a total of 29 requests. </b>
<br /> <br />
<b> Now I will create a chart in Tableau to display for the above results. </b>

In [11]:
chart = 'C:/Users/Rickardo.Henry/Desktop/IS_362_FinalProject/Data%20Files/Complaint%20Types.csv'
chart_ser_req = pd.read_csv(chart)
chart_ser_req.head()

Complaint Types   Totals 
0  Illegal Parking                     29
1     Blocked Driveway                 24
2     Noise - Street/Sidewalk          22
3      Street Condition                12
4         Noise - Commercial            9

<b>I used the results above to create a chart <i>(.csv file)</i> and saved the results.</b>
<br /><br />
<b> Take a look at the</b> [Complaint Types Chart](https://public.tableau.com/profile/erica3118#!/vizhome/ComplaintTypes/Sheet1?publish=yes) <b>to view the chart in Tableau.</b> 

<b> From the chart we are able to view that Illegal Parking was the service request taken most in 2015.</b>
<br /> <br />

<b> We now can continue to answer the next question. <br /><br />
Question #2: Based on the data given, what season recieved the most service requests in 2015 ? </b>

In [12]:
date = ser_req.groupby('Created Date').count()
date.head()

Unique Key  Closed Date  Agency  Agency Name  \
Created Date                                                     
10/1/2015 7:02             1            1       1            1   
10/19/2015 13:32           1            1       1            1   
10/19/2015 7:51            1            1       1            1   
10/8/2015 10:09            1            1       1            1   
10/8/2015 13:12            2            2       2            2   

                  Complaint Type  Descriptor  Location Type  Incident Zip  \
Created Date                                                                
10/1/2015 7:02                 1           1              1             1   
10/19/2015 13:32               1           1              0             1   
10/19/2015 7:51                1           1              1             1   
10/8/2015 10:09                1           1              1             1   
10/8/2015 13:12                2           2              2             2   

                  Incident Address  City  Status  Resolution Description  \
Created Date                                                               
10/1/2015 7:02                   1     1       1                       1   
10/19/2015 13:32                 1     1       1                       1   
10/19/2015 7:51                  1     1       1                       1   
10/8/2015 10:09                  1     1       1                       1   
10/8/2015 13:12                  2     2       2                       2   

                  Borough  
Created Date               
10/1/2015 7:02          1  
10/19/2015 13:32        1  
10/19/2015 7:51         1  
10/8/2015 10:09         1  
10/8/2015 13:12         2

<b> Based on the results, the date of October 8th recieved the most service requests. It got a total of 2 complaints for that day. With such conclusions, we can say that the FALL season recieved the most service requests in 2015 as October is in the Fall season. </b>

<b> Take a look at the</b> [SEASON'S CHART](https://public.tableau.com/profile/erica3118#!/vizhome/DateChart/Sheet1?publish=yes) <b>.</b> <br /> <br />
<b> Now let's move unto the next question. <br />  
Q3: What borough submitted the most service requests in 2015? </b>

In [13]:
borough = ser_req.groupby('Borough').count()
borough.head()

Unique Key  Created Date  Closed Date  Agency  Agency Name  \
Borough                                                                     
BRONX                  29            29           27      29           29   
BROOKLYN               40            40           39      40           40   
MANHATTAN              30            30           29      30           30   
QUEENS                 40            40           38      40           40   
STATEN ISLAND           6             6            6       6            6   

               Complaint Type  Descriptor  Location Type  Incident Zip  \
Borough                                                                  
BRONX                      29          29             29            29   
BROOKLYN                   40          40             40            40   
MANHATTAN                  30          28             28            30   
QUEENS                     40          40             35            39   
STATEN ISLAND               6           6              6             6   

               Incident Address  City  Status  Resolution Description  
Borough                                                                
BRONX                        22    29      29                      28  
BROOKLYN                     31    40      40                      40  
MANHATTAN                    20    30      30                      30  
QUEENS                       33    39      40                      40  
STATEN ISLAND                 4     6       6                       6

<b> From the results we can see that BROOKLYN and QUEENS were the two boroughs with most service requests in 2015 with a total of 40 requests each.</b> <br /><br /> <b> Again, we can view the results in the</b> [Borough Service Request Chart](https://public.tableau.com/profile/erica3118#!/vizhome/BoroughServiceRequests-2015/Sheet1?publish=yes) 
<br /><br />
<b> We can proceed unto the next question. </b> <br /><br />
<b> Question #3: What what the total of service requests in 2015? </b>

In [14]:
ser_req['Complaint Type'].count()

149

<b> There were a total of 149 service requests in 2015. But how many of those are closed & how many remain open? </b>

In [15]:
ser_req['Status'].value_counts()

Closed     143
Open         5
Pending      1
Name: Status, dtype: int64

<b> There are 143 CLOSED and 5 service request which remained OPEN. ONE service request remained pending. Let's take a look at the</b> [2015 - Total Service Requests Chart](https://public.tableau.com/profile/erica3118#!/vizhome/TotalServiceRequests-2015/Sheet1?publish=yes)

<h1> Conclusion: </h1> <br /> 
    <b>My biggest challenges were how to add an image to my project and as well how to work with dates & times to calculate the seasons. I eventually was able to figure out how to add an image which was exciting for me. At first, I planned to use a pie chart and other charts to display my data, but went back to using Tableau to keep it fresh in my mind as well I think it is easier on the eye for me.</b> 

<div align="center"> <b> <span style='color:green;'> <h1> THANK YOU !</h1> </span> </b> </div>